packages/cloud_firestore_odm/cloud_firestore_odm_generator/lib/src/templates/query_reference.dart

```diff
@@ -401,3 +403,12 @@ class ${data.queryReferenceImplName}
 ''';
   }
 }
+
+extension on TypeProvider {
+  DartType asNullable(DartType type) {
+    final typeSystem = nullType.element.library.typeSystem;
+    if (typeSystem.isNullable(type)) return type;
+
+    return typeSystem.leastUpperBound(type, nullType);
```

Comment:
What does it do?

Reply:
leastUpperBound returns the closest shared type between the two parameters

For `leastUpperBound(int, double)` that'd be `num`. For ``leastUpperBound(int, String)` that'd be `Object`

And for `leastUpperBound(int, null)` we'll get `int?`

Reply:
Ohh nice thanks :)